lib/services/statsRegistry.js

```diff
@@ -0,0 +1,192 @@
+    timerHandler,
+    statsContext = {
+        op: 'IoTAgentNGSI.TimedStats'
+    };
+
+/**
+ * Add a new stat value to the specified key of the stats registry. The stat is added
+ * both to the global and current stats.
+ *
+ * @param {String} key          Name of the stat that is going to be written.
+ * @param {Number} value        Value to be added to the total.
+ */
+function add(key, value, callback) {
+    if (currentStats[key]) {
+        currentStats[key] += value;
+    } else {
+        currentStats[key] = value;
+    }
+
+    if (globalStats[key]) {
+        globalStats[key] += value;
+    } else {
+        globalStats[key] = value;
+    }
+
+    callback(null);
+}
+
+/**
+ * Get the current value of a particular stat.
+ *
+ * @param {String} key          Name of the stat to retrive.
+ */
+function getCurrent(key, callback) {
+    callback(null, currentStats[key]);
+}
+
+/**
+ * Get the global value of the selected attribute.
+ *
+ * @param {String} key          Name of the stat to retrive.
+ */
+function getGlobal(key, callback) {
+    callback(null, globalStats[key]);
+}
+
+/**
+ * Get all the global stats currently stored in the repository.
+ */
+function getAllGlobal(callback) {
+    callback(null, globalStats);
+}
+
+/**
+ * Get all the current stats currently stored in the repository.
+ */
+function getAllCurrent(callback) {
+    callback(null, currentStats);
+}
+
+/**
+ * Loads the values passed as parameters into the global statistics repository.
+ *
+ * @param {Object} values       Key-value map with the values to be load.
+ */
+function globalLoad(values, callback) {
+    globalStats = values;
+    currentStats = {};
+
+    for (var i in values) {
+        if (values.hasOwnProperty(i)) {
+            currentStats[i] = 0;
+        }
+    }
+
+    callback(null);
+}
+
+/**
+ * Changes the configuration of the Stats Registry service.
+ *
+ * @param {Object} newConfig        New configuration of the registry.
+ */
+function init(newConfig, callback) {
+    config = newConfig;
+
+    if (callback) {
+        callback();
+    }
+}
+
+/**
+ * Reset each of the current stats to value zero.
+ */
+function resetCurrent(callback) {
+    for (var i in currentStats) {
+        if (currentStats.hasOwnProperty(i)) {
+            currentStats[i] = 0;
+        }
+    }
+
+    callback();
+}
+
+/**
+ * Executes all the stored timer actions when a timer click is received.
+ */
+function tickHandler() {
+    async.series(timerActions);
+}
+
+/**
+ * Adds a new timer action to the timerActions Array, activating the timer if it was not previously activated.
+ *
+ * @param {Function} handler        Action to be executed. Should take two statistics objects and a callback.
+ */
+function addTimerAction(handler, callback) {
+    if (!timerHandler && config.stats.interval) {
+        timerHandler = setInterval(tickHandler, config.stats.interval);
+    }
+
+    timerActions.push(apply(handler, currentStats, globalStats));
+    callback();
```

Comment:
Although I guess you are the only one using this library for the time being :) , what about protecting the invocation of this `addTimerAction()` function by third parties checking that `callback` is really passed, I mean with something such as:
```javascript
if (callback) {
  callback();
}
```
I leave it to your discretion... :)